manager/manager.go

```diff
@@ -433,6 +434,9 @@ func (m *Manager) Run(parent context.Context) error {
 	}
 
 	cs := raftpicker.NewConnSelector(m.RaftNode, proxyOpts...)
+	m.mu.Lock()
+	m.connSelector = cs
```

Comment:
If `Stop` is called very quickly, before `Run` gets here, we would skip shutting down the ConnSelector (and it might panic when the raft node is destroyed).

Maybe we should close a channel after `Run` does this assignment, and `Stop` should wait for that channel as the first thing it does?